### PR TITLE
Various mkosi improvements

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -37,6 +37,9 @@ indent_size = 2
 indent_style = space
 max_line_length = 109
 
+[*.md]
+max_line_length = 80
```
Comment (yuwata) [inline .editorconfig]: I don't think it is necessary. Why 109 is a bit mach?

File: .github/workflows/mkosi.yml [hunk 2]
```diff
@@ -153,8 +153,8 @@ jobs:
           [Build]
           UseSubvolumes=yes
           ToolsTree=default
-          ToolsTreeDistribution=fedora
-          ToolsTreeRelease=rawhide
+          ToolsTreeDistribution=${{ matrix.distro }}
+          ToolsTreeRelease=${{ matrix.release }}
```
Comment (yuwata) [inline .github/workflows/mkosi.yml]: Does it now work? When I asked previously, you said it does not work.

**Review by yuwata — CHANGES_REQUESTED** [review]
Looks basically good. But I do not like the 80ch change. Also one question.

File: .github/workflows/mkosi.yml (hunk 2 above)
Comment (daandemeyer) [reply .github/workflows/mkosi.yml]: It doesn't work for all distros, but I added the change to figure out what doesn't work, I'll drop it again to get CI to pass

**Review by yuwata — APPROVED** [review]

File: .github/workflows/mkosi.yml (hunk 2 above)
Comment (daandemeyer) [reply .github/workflows/mkosi.yml]: So opensuse has a weird build failure that I can't look into until https://build.opensuse.org/request/show/1232687 is fixed. CentOS 9/10 don't have `btrfs` which we need to make the tests not go over the limited disk size. And on Ubuntu `meson` is too old and doesn't have `--max-lines`.

**Comment by bluca** [comment]
What's the arcana to disable the tools tree in mkosi.local.conf?

**Comment by DaanDeMeyer** [comment]
> What's the arcana to disable the tools tree in mkosi.local.conf?

It's literally in the documentation I wrote in this PR

**Comment by bluca** [comment]
> > What's the arcana to disable the tools tree in mkosi.local.conf?
> 
> It's literally in the documentation I wrote in this PR

In my defence, the commit said "simplify doc" and was a 300 lines diff, so missed it :-P

**Comment by evverx** [comment]
Based on feedback like https://github.com/systemd/systemd/issues/34885#issuecomment-2499197178 and https://github.com/systemd/systemd/pull/34888#issuecomment-2541439039 I'm not sure `mkosi sandbox` can significantly improve the experience. (The PR had to be rebased on top of the main branch three times to get mkosi to work as far as I can remember but it's nice that the stuff tested by the CI is recommended).

**Comment by yuwata** [comment]
> Based on feedback like [#34885 (comment)](https://github.com/systemd/systemd/issues/34885#issuecomment-2499197178) and [#34888 (comment)](https://github.com/systemd/systemd/pull/34888#issuecomment-2541439039) I'm not sure `mkosi sandbox` can significantly improve the experience. (The PR had to be rebased on top of the main branch three times to get mkosi to work as far as I can remember but it's nice that the stuff tested by the CI is recommended).

Requiring rebase during working on is quite often, not only by changing the test framework. And we are actively updating testing framework now. So, that's expected and we cannot and should not stop the way, even if you or the author or the PR feels that's annoying.

**Comment by evverx** [comment]
@yuwata it has nothing to do with the testing framework. That stuff was used to test networkd manually without running any integration tests.

**Comment by yuwata** [comment]
testing-farm:fedora-rawhide fails with the following:
```
>>> Running trigger-install scriptlet: systemd-0:257-1.20241220125434840053.pr35
>>> Finished trigger-install scriptlet: systemd-0:257-1.20241220125434840053.pr3
>>> Scriptlet output:
>>> Creating group 'systemd-coredump' with GID 994.
>>> Creating user 'systemd-coredump' (systemd Core Dumper) with UID 994 and GID 
>>> Creating group 'systemd-timesync' with GID 993.
>>> Creating user 'systemd-timesync' (systemd Time Synchronization) with UID 993
>>> 
Warning: skipped OpenPGP checks for 407 packages from repositories: test-artifacts, testing-farm-tag-repository
Complete!
‣  Running prepare script /var/tmp/tmp.Dyoj18rTfm/systemd/mkosi.conf.d/10-centos-fedora/mkosi.prepare…
No matches were found for the following plugin name patterns while enabling libdnf plugins: builddep, versionlock
No matches were found for the following plugin name patterns while disabling libdnf plugins: *
No matches were found for the following plugin name patterns while enabling libdnf plugins: builddep, versionlock
No matches were found for the following plugin name patterns while disabling libdnf plugins: *
Updating and loading repositories:
Cache-only enabled but no cache for repository "rawhide"
‣ "/work/prepare final" returned non-zero exit code 123.
FAILED: . 
/var/tmp/tmp.Dyoj18rTfm/systemd/../mkosi/bin/mkosi --directory /var/tmp/tmp.Dyoj18rTfm/systemd --output-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.output --cache-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.cache --build-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.builddir --extra-search-path /var/tmp/tmp.Dyoj18rTfm/systemd/build --force build
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /var/tmp/tmp.Dyoj18rTfm/systemd/build .
```
Is this related? Or, an issue in rawhide repository?

**Comment by yuwata** [comment]
> @yuwata it has nothing to do with the testing framework. That stuff was used to test networkd manually without running any integration tests.

I have not followed the linked comments, but respond to the following part of your comment:
> (The PR had to be rebased on top of the main branch three times to get mkosi to work as far as I can remember but it's nice that the stuff tested by the CI is recommended)

And still not followed the linked comments, but systemd-networkd-tests.py can run on local machine without mkosi.

**Comment by evverx** [comment]
@yuwata I don't remember the details but it was a relatively painful process. The first attempt was on Debian Stable and it didn't work of course. Then it was tried on Debian Unstable and it didn't work either. Debian Testing got better but some stuff failed so the PR had to be rebased. Then it worked fine for a while and broke again and had to be rebased again and so on. The point I'm trying to make is that the process isn't as smooth as it can be (partly because all those things aren't mentioned)

**Comment by yuwata** [comment]
@evverx Again, unfortunately that's usual. Failures might be caused by broken GitHub infra, broken distro repository, broken dependent libraries, broken mkosi, broken code, and so on. You need to watch everything, or just wait for a while.

Anyway, that's completely off topic for this PR. Please request more on #35492 if you have any better ideas.

**Comment by DaanDeMeyer** [comment]
> testing-farm:fedora-rawhide fails with the following:
> 
> ```
> >>> Running trigger-install scriptlet: systemd-0:257-1.20241220125434840053.pr35
> >>> Finished trigger-install scriptlet: systemd-0:257-1.20241220125434840053.pr3
> >>> Scriptlet output:
> >>> Creating group 'systemd-coredump' with GID 994.
> >>> Creating user 'systemd-coredump' (systemd Core Dumper) with UID 994 and GID 
> >>> Creating group 'systemd-timesync' with GID 993.
> >>> Creating user 'systemd-timesync' (systemd Time Synchronization) with UID 993
> >>> 
> Warning: skipped OpenPGP checks for 407 packages from repositories: test-artifacts, testing-farm-tag-repository
> Complete!
> ‣  Running prepare script /var/tmp/tmp.Dyoj18rTfm/systemd/mkosi.conf.d/10-centos-fedora/mkosi.prepare…
> No matches were found for the following plugin name patterns while enabling libdnf plugins: builddep, versionlock
> No matches were found for the following plugin name patterns while disabling libdnf plugins: *
> No matches were found for the following plugin name patterns while enabling libdnf plugins: builddep, versionlock
> No matches were found for the following plugin name patterns while disabling libdnf plugins: *
> Updating and loading repositories:
> Cache-only enabled but no cache for repository "rawhide"
> ‣ "/work/prepare final" returned non-zero exit code 123.
> FAILED: . 
> /var/tmp/tmp.Dyoj18rTfm/systemd/../mkosi/bin/mkosi --directory /var/tmp/tmp.Dyoj18rTfm/systemd --output-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.output --cache-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.cache --build-dir /var/tmp/tmp.Dyoj18rTfm/systemd/build/mkosi.builddir --extra-search-path /var/tmp/tmp.Dyoj18rTfm/systemd/build --force build
> ninja: build stopped: subcommand failed.
> INFO: autodetecting backend as ninja
> INFO: calculating backend command to run: /usr/bin/ninja -C /var/tmp/tmp.Dyoj18rTfm/systemd/build .
> ```
> 
> Is this related? Or, an issue in rawhide repository?

I'm not sure, I opened https://github.com/systemd/systemd/pull/35700 for starters to move this stuff into upstream.

**Comment by evverx** [comment]
> unfortunately that's usual

I'm aware of that but people new to systemd aren't.

> that's completely off topic for this PR

As far as I understand the goal of the HACKING commit is to make it easier for new contributors to develop systemd so I wouldn't say it's completely off-topic. I just pointed out that what it looks like when new contributors try to follow the documentation.

**Comment by bluca** [comment]
```
5722s ‣ (Make sure the right keyring package (e.g. debian-archive-keyring, kali-archive-keyring or ubuntu-keyring) is installed)
5722s FAILED: . 
```

**Pull request merged**